public/JavaScript/gallery.js

```diff
@@ -62,4 +62,38 @@ function reportGalleryStats (imagesInDom) {
   return String(imagesInDom)
 }
 
-module.exports = reportGalleryStats
+/**
+ *
+ * @param {Array} userSelectedFilters
+ * @returns {Object} activeFilters
+ */
+function generateFilterOptions (userSelectedFilters) {
+  const allowedFilters = {
+    gender: ['male', 'female'],
+    type: ['dog', 'cat', 'fox', 'parrot'],
+    rating: [1, 2, 3, 4, 5]
+  }
+
+  const activeFilters = {
+    gender: null,
+    type: null,
+    rating: null
+  }
+
+  Object.keys(userSelectedFilters).forEach(filterKey => {
+    if (Object.prototype.hasOwnProperty.call(allowedFilters, filterKey)) {
+      const filterValue = userSelectedFilters[filterKey]
+      // Only proceed if filterValue is an array
+      if (Array.isArray(filterValue)) {
+        const lastFilterValue = filterValue[filterValue.length - 1]
+        // Check if the last value of the array is an allowed value
+        activeFilters[filterKey] = allowedFilters[filterKey].includes(lastFilterValue) ? lastFilterValue : null
+      } else {
```

Comment:
Transform this into a guard block.

Start the function by checking all conditions. Start with the known failing cases. Return early. 

If it passes each guard then at the end of the function it reaches it means that all conditions are met do something you are supposed to do